### PR TITLE
Compile bundled files and typing files

### PR DESCRIPTION
**Pull request by maier49**
This PR adds a flag that modifies the behavior of `ts-loader` so that it only loads files that are bundled by webpack and any `.d.ts` files that match the config in `tsconfig.json`. This is an attempt to implement the behavior described in #267, or at least a starting point. 

The approach is very simple and I can't help but feel like there's some obvious issue with it that I'm not thinking of, but it seems to work fine in the test case I created at least. The reason for compiling all `.d.ts` files as well as bundled files is that they may include types that are used in files that are bundled without being explicitly imported. Attempting to determine exactly which files are needed seems like it may be quite difficult and potentially slow.

### CODE REVIEW

**Comment by johnnyreilly** [comment]
Awesome - thanks for this! I'll try and take a look later on. BTW we only need comparison test output for the latest released version of TypeScript (2.5 right now). I really should make that clearer in the docs!

File: src/index.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ function getLoaderOptions(loader: Webpack) {
 }
 
 type ValidLoaderOptions = keyof LoaderOptions;
-const validLoaderOptions: ValidLoaderOptions[] = ['silent', 'logLevel', 'logInfoToStdOut', 'instance', 'compiler', 'configFile', 'transpileOnly', 'ignoreDiagnostics', 'errorFormatter', 'colors', 'compilerOptions', 'appendTsSuffixTo', 'appendTsxSuffixTo', 'entryFileCannotBeJs' /* DEPRECATED */, 'happyPackMode', 'getCustomTransformers'];
+const validLoaderOptions: ValidLoaderOptions[] = ['silent', 'logLevel', 'logInfoToStdOut', 'instance', 'compiler', 'configFile', 'transpileOnly', 'ignoreDiagnostics', 'errorFormatter', 'colors', 'compilerOptions', 'appendTsSuffixTo', 'appendTsxSuffixTo', 'entryFileCannotBeJs', 'onlyCompileBundledFiles' /* DEPRECATED */, 'happyPackMode', 'getCustomTransformers'];
```
Comment (johnnyreilly) [inline src/index.ts]: Please can you move `/* DEPRECATED */` back to after `'entryFileCannotBeJs`?

**Comment by johnnyreilly** [comment]
Generally this looks good - could you remove comparison test output apart from 2.5?  Also could you add something to the `README` about the new option?

**Comment by maier49** [comment]
@johnnyreilly Thanks for the quick review. I made some updates to address your comments.

**Comment by johnnyreilly** [comment]
Sweet - I'll try and take a look tomorrow. Thanks!

**Comment by johnnyreilly** [comment]
Perfect!

**Comment by johnnyreilly** [comment]
Hope to ship this with ts-loader 3.1.0

**Pull request merged**